src/sprite.c

```diff
@@ -152,406 +146,118 @@ void sub_8004E8C(unkStruct_2039DB0 *a0)
+    *(_dstVar++) = (*(_srcVar++) & _spriteMasks->unk2) | _spriteMasks->unk8;    \
+    *(_dstVar++) = (*(_srcVar++) & _spriteMasks->unk4) | _spriteMasks->unkA;    \
+    *(_dstVar++) = *(_srcVar++);                                                \
 }
 
-// https://decomp.me/scratch/VYqKb / https://decomp.me/scratch/33KIb - Same issue as AddSprite()
-// Should be functionally equivalent.
-#ifdef NONMATCHING
 void AddAxSprite(ax_pose *a0, axdata1 *a1, UnkSpriteMem *a2, unkStruct_2039DB0 *spriteMasks)
 {
     // size: 0xC
     struct UnkStackFor8004EA8
     {
-        u16 unk0;
+        s16 unk0;
         ax_pose_unk2 unk2;
         SpriteOAM oam;
     } sp;
     SpriteOAM *sprite;
-    s32 r7;
-    s32 x;
-    s32 y;
+    s32 spriteId;
+    s32 pos; // Has to be used for both x and y to match
+    struct UnkStackFor8004EA8 *spPtr;
 
     if (a2 != NULL)
         RegisterSpriteParts_80052BC(a2);
 
     if (sSpriteCount >= 128)
         return;
 
-    sp.unk0 = a0->sprite;
-    sp.unk2 = a0->unk2;
-    sp.oam.attrib1 = a0->flags1 & ~(0x100 | 0x200);
-    sp.oam.attrib2 = a0->flags2 & ~(0x200 | 0x400 | 0x800);
-    sp.oam.attrib3 = a0->flags3;
-    sp.oam.unk6 = ((a0->flags2 & (0x200 | 0x400 | 0x800)) >> 9) | ((a0->flags1 & (0x1 | 0x2 | 0x4 | 0x8 | 0x10 | 0x20 | 0x40 | 0x80 | 0x100 | 0x200)) << 4);
-    sprite = sUnknown_20262A8 + sSpriteCount;
-    r7 = a1->unk16 + sp.unk2.unk1;
+    spPtr = &sp;
+    spPtr->unk0 = a0->sprite;
+    spPtr->unk2 = a0->unk2;
+
+    spPtr->oam.attrib1 = a0->flags1 & ~(0x100 | 0x200);
+    spPtr->oam.attrib2 = a0->flags2 & ~(0x200 | 0x400 | 0x800);
+    spPtr->oam.attrib3 = a0->flags3;
+    spPtr->oam.unk6 = ((a0->flags2 & (0x200 | 0x400 | 0x800)) >> 9) | ((a0->flags1 & (0x1 | 0x2 | 0x4 | 0x8 | 0x10 | 0x20 | 0x40 | 0x80 | 0x100 | 0x200)) << 4);
+    sprite = &sUnknown_20262A8[sSpriteCount];
+    spriteId = a1->unk16 + sp.unk2.unk1;
 
-    if (r7 < 0)
-        r7 = 0;
-    if (r7 > 255)
-        r7 = 255;
+    if (spriteId < 0)
+        spriteId = 0;
+    if (spriteId > 255)
+        spriteId = 255;
 
     if (spriteMasks == NULL) {
-        CopySprite(sprite, &sp.oam);
+        SpriteCopy(sprite, &spPtr->oam);
     }
     else {
-        CopySpriteWithMasks(sprite, &sp.oam, spriteMasks);
+        SpriteCopyWithMasks(sprite, &spPtr->oam, spriteMasks);
     }
 
     // Set tileNum
     if (sp.unk2.unk0 != 0 && sUnknown_2025672[sp.unk2.unk0] != 0) {
-        s32 tileNum = sUnknown_2025672[sp.unk2.unk0];
-        tileNum &= SPRITEOAM_MAX_TILENUM;
-        tileNum <<= SPRITEOAM_SHIFT_TILENUM;
-        sprite->attrib3 &= ~SPRITEOAM_MASK_TILENUM;
-        sprite->attrib3 |= tileNum;
+        SpriteSetTileNum(sprite, sUnknown_2025672[sp.unk2.unk0]);
     }
     else {
         // Animations add to existing tileNum
-        s32 newTileNum;
         s32 tileNum = sprite->attrib3;
         tileNum &= SPRITEOAM_MAX_TILENUM;
         tileNum >>= SPRITEOAM_SHIFT_TILENUM;
```

Comment:
This order of operations seems wrong. Should be `>>` first and then the `&`. But perhaps this can be made into a `SpriteGetTileNum()`?